src/shared/ui/inputPrompter.ts

```diff
@@ -84,19 +102,19 @@ export class InputBoxPrompter extends Prompter<string> {
      * Sets a validation hook into the InputBox, checking whenever the input changes or when the user
      * attempts to submit their response.
      *
-     * @param validate Validator function
+     * @param validate Validator function.
      */
-    public setValidation(validate: (value: string) => string | undefined): void {
+    public setValidation(validate: ValidateFn): void {
         this.validateEvents.forEach(d => d.dispose())
         this.validateEvents = []
 
         this.inputBox.onDidChangeValue(
-            value => (this.inputBox.validationMessage = validate(value)),
+            value => (this.inputBox.validationMessage = validate(value, false)),
```

Comment:
Reverted the Thenable change https://github.com/aws/aws-toolkit-vscode/pull/3114/commits/7d1c81c8203ea2dfbedddf5e85f38a95e5f08f76 for now since it doesn't work with this `inputBox.onDidChangeValue` and `inputBox.onDidAccept`.